lib/homekit-device.js

```diff
@@ -3156,7 +3157,24 @@ class HomeKitDevice extends Device {
         this.connectionType
       );
 
-      return client.pairSetup(pin).then(() => {
+      // special case where the PIN will be displayed to the user on-screen
+      if (pin === '000-00-000') {
+        return client.startPairing().then((data) => {
+          this.pendingPairingData = data;
+          throw new Error('Enter new PIN from device\'s display');
+        });
+      }
+
+      let promise;
+      if (this.pendingPairingData) {
+        promise = client.finishPairing(this.pendingPairingData, pin);
```

Comment:
This isn't working currently because we use a different `client` for start/finish. It needs to be the same one. Maybe just set `this.client` above when calling `startPairing()`?

Reply:
Good call. Just pushed a fix.